src/com/amazon/corretto/crypto/provider/RsaCipher.java

```diff
@@ -211,25 +236,129 @@ protected AlgorithmParameters engineGetParameters() {
     }
 
     @Override
-    protected void engineInit(final int opmode, final Key key, final SecureRandom random)
-            throws InvalidKeyException {
+    protected void engineInit(final int opmode, final Key key, final AlgorithmParameterSpec params,
+            final SecureRandom random)
+                    throws InvalidKeyException, InvalidAlgorithmParameterException {
+        if (params != null) {
+            if (params instanceof OAEPParameterSpec) {
+                if (padding_ != Padding.OAEP) {
```

Comment:
I think that this means the following code is valid when it should throw an exception:

```java
Cipher c = Cipher.getInstance("RSA/ECB/OAEPWithSHA-1AndMGF1Padding");
c.init(Cipher.ENCRYPT_MODE, key, new OAEPParameterSpec("SHA-256", "MGF1", MGF1ParameterSpec.SHA256, PSource.PSpecified.DEFAULT)); // Should throw InvalidAlgorithmParameterException
```

Reply:
ah, yes you're right. i'd thought that this was also a case where we should adjust the OAEP parameters based on the passed OAEPParameterSpec. if we're allowing for the more general `RSA/ECB/OAEP`, then i see that we'd want to treat `RSA/ECB/OAEPWithSHA-1AndMGF1Padding` as "fixed".

i'll update the code and tests to handle this correctly.